clang/test/SemaSYCL/array-kernel-param.cpp

```diff
@@ -230,3 +256,32 @@ int main() {
 // CHECK-NEXT: DeclRefExpr {{.*}} 'int':'int'
 // CHECK-NEXT: ImplicitCastExpr {{.*}} 'int':'int'
 // CHECK-NEXT: DeclRefExpr {{.*}} 'int':'int'
+
+// Check kernel_F parameters
+// CHECK: FunctionDecl {{.*}}kernel_F{{.*}} 'void (int, int, int, int, int, int)'
+// CHECK-NEXT: ParmVarDecl {{.*}} used _arg_ 'int'
+// CHECK-NEXT: ParmVarDecl {{.*}} used _arg_ 'int'
+// CHECK-NEXT: ParmVarDecl {{.*}} used _arg_ 'int'
+// CHECK-NEXT: ParmVarDecl {{.*}} used _arg_ 'int'
+// CHECK-NEXT: ParmVarDecl {{.*}} used _arg_ 'int'
+// CHECK-NEXT: ParmVarDecl {{.*}} used _arg_ 'int'
+// Check kernel_F inits
+// CHECK-NEXT: CompoundStmt
+// CHECK-NEXT: DeclStmt
+// CHECK-NEXT: VarDecl {{.*}} cinit
+// CHECK-NEXT: InitListExpr
+// CHECK-NEXT: InitListExpr {{.*}} 'int [2][3]'
+// CHECK-NEXT: InitListExpr {{.*}} 'int [3]'
```

Comment:
Same question here.   Shouldn't this be 'int [2]'?

Reply:
No it should be int[3] because it is 2 int [3] arrays.